### ISSUE TITLE
[SR-13477] Cannot build toolchain with Xcode 12 beta 3 or later.

### ISSUE THREAD

**Issue by johnno1962** [swiftlang/llvm-project#4421]
|                  |                 |
   |------------------|-----------------|
   |Previous ID       | SR-13477      |
   |Radar             | rdar://problem/68115900         |
   |Original Reporter | @johnno1962      |
   |Type              | Bug    |

<details>
  <summary>Environment</summary>

Catalina 10.15.4, Xcode 12 beta 2

</details>

   <details>
  <summary>Additional Detail from JIRA</summary>

   |                  |                 |
   |------------------|-----------------|
   |Votes             | 0         |
   |Component/s       | Compiler, LLDB for Swift    |
   |Labels            | Bug        |
   |Assignee          | None      |
   |Priority          | Medium      |

   

   md5: b5ad3e2c5b42677f9f4191283388a8d7

  </details>

**Issue Description:**

Hi Apple,

I've been re-syncing my swift repo and found I couldn't build a toolchain with beta 6 or beta 3 (as it suggests in the project README) giving an error building part of lldb saying \_main undefined for architecture arm64. I've just successfully completed a build using beta 2 (the last beta before Apple Silicon). Should the README be updated or is there a specific reason why beta 3 is listed as the preferred version?

**Comment by LucianoPAlmeida** [comment]
@johnno1962 I had the same issue moving from beta 1 to beta 6, to build successfully you have to pass the --swift-darwin-supported-archs 'x86_64' flag on the build-script.  
Perhaps this should have to be be in the docs somewhere

**Comment by johnno1962** [comment]
Thanks! I guess it should be.

**Comment by typesanitizer** [comment]
@swift-ci create

**Comment by edymtt** [comment]
I expect this particular issue not happening -- at the time we had still a few regressions regarding building on Apple Silicon, now we expect to be able to build out of the box with recent Xcodes.

Also the [documentation](https://github.com/swiftlang/swift/blob/main/docs/HowToGuides/GettingStarted.md#the-actual-build) has been updated to reflect the need to pass `--swift-darwin-supported-archs` when building at desk.

If similar issues keep happening, please file a new issue (as this one refers to an old configuration that we no longer support).

**Comment by johnno1962** [comment]
Yes, this is ancient history now but see also https://github.com/swiftlang/swift/pull/87593
